tests/test_check_parallel.py

```diff
@@ -0,0 +1,305 @@
+        run_cmd('{}zstash check --workers=2'.format(zstash_dir))
+        actual = sorted(os.listdir('zstash'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+        shutil.rmtree('zstash')
+
+    def testKeepTarsWithPreviouslySetHPSS_2(self):
+        """
+        Test `zstash check` in parallel when hpss is set in `zstash create`.
+
+        This time using self.test_dir (which is set as 'zstash_test' in base.setUp(),
+        but we'll keep it 'zstash', as in _1). Passes.
+
+        Bash commands:
+        # In zstash/tests directory
+        mkdir zstash
+        zstash create --hpss=none --maxsize 128 .
+        ls zstash
+        zstash check --workers=2
+        ls zstash
+        """
+        print_in_box('testKeepTarsWithPreviouslySetHPSS')
+        self.assertWorkspace()
+        self.test_dir = 'zstash'
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+        os.mkdir(self.test_dir)
+        zstash_dir = ZSTASH_DIR
+        self.hpss_path = 'none'
+        # Run `zstash create`
+        run_cmd('{}zstash create --hpss={} --maxsize 128 .'.format(zstash_dir, self.hpss_path))
+        actual = sorted(os.listdir('zstash'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+        # Run `zstash check` without specifying hpss
+        run_cmd('{}zstash check --workers=2'.format(zstash_dir))
+        actual = sorted(os.listdir('zstash'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+
+    def testKeepTarsWithPreviouslySetHPSS_3(self):
+        """
+        Test `zstash check` in parallel when hpss is set in `zstash create`.
+
+        This time using self.test_dir using the default of 'zstash_test' 
+        (set in base.setUp()). Passes.
+
+        Bash commands:
+        # In zstash/tests directory
+        mkdir zstash_test
+        zstash create --hpss=none --maxsize 128 .
+        ls zstash
+        zstash check --workers=2
+        ls zstash
+        """
+        print_in_box('testKeepTarsWithPreviouslySetHPSS')
+        self.assertWorkspace()
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+        os.mkdir(self.test_dir)
+        zstash_dir = ZSTASH_DIR
+        self.hpss_path = 'none'
+        # Run `zstash create`
+        run_cmd('{}zstash create --hpss={} --maxsize 128 .'.format(zstash_dir, self.hpss_path))
+        actual = sorted(os.listdir('zstash'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+        # Run `zstash check` without specifying hpss
+        run_cmd('{}zstash check --workers=2'.format(zstash_dir))
+        actual = sorted(os.listdir('zstash'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+
+    def testKeepTarsWithPreviouslySetHPSS_4(self):
```

Comment:
@golaz I made 4 more versions of `testKeepTarsWithPreviouslySetHPSS`. `_1` is the original; I think `_4` is how I want the test to look. However, `_4` and `_5` fail -- it's as if this test can only be run from the `zstash/tests` directory. I added descriptions of each version in the corresponding doc-string, along with a list of the equivalent bash commands. You can compare the exact code of each function by using a diff checker like https://www.diffchecker.com/.